shared/src/business/useCases/users/updateUserContactInformationInteractor.test.ts

```diff
@@ -227,6 +232,55 @@ describe('updateUserContactInformationInteractor', () => {
     expect(generateChangeOfAddress).toHaveBeenCalled();
   });
 
+  it('should clean up DB and send websocket message if "generateChangeOfAddress" returns empty array', async () => {
+    (generateChangeOfAddress as jest.Mock).mockReturnValue([]);
+
+    await updateUserContactInformationInteractor(applicationContext, {
+      contactInfo,
+      userId: mockUser.userId,
+    } as any);
+
+    expect(
+      applicationContext.getPersistenceGateway().updateUser.mock.calls[1][0]
+        .isUpdatingInformation,
+    ).not.toBeDefined();
+
+    const notificatsionCalls =
+      applicationContext.getNotificationGateway().sendNotificationToUser.mock
+        .calls;
+
+    expect(
+      notificatsionCalls[notificatsionCalls.length - 1][0].message.action,
+    ).toEqual('user_contact_full_update_complete');
+
+    expect(
+      notificatsionCalls[notificatsionCalls.length - 1][0].message.user,
+    ).toMatchObject({
+      contact: contactInfo,
+    });
+  });
+
+  it('should not clean up DB and send websocket message if "generateChangeOfAddress" returns undefined', async () => {
+    (generateChangeOfAddress as jest.Mock).mockReturnValue(undefined);
+
+    await updateUserContactInformationInteractor(applicationContext, {
+      contactInfo,
+      userId: mockUser.userId,
+    } as any);
```

Comment:
Just curious why this test / use case is permitted? I recognize that `generateChangeOfAddress` should never return undefined, but I don't understand the rationale that we leave the user's account in a broken state if something like this should happen. Why not send the websocket message and clean up the DB regardless?